opentelemetry-sdk/src/opentelemetry/sdk/_metrics/metric_reader_storage.py

```diff
@@ -55,50 +51,46 @@ def _get_or_init_view_instrument_match(
                 return self._view_instrument_match[instrument]
 
             # not present, hold the lock and add a new mapping
-            matches = []
+            view_instrument_matches = []
             for view in self._sdk_config.views:
-                if view.match(instrument):
-                    # Note: if a view matches multiple instruments, this will create a separate
-                    # _ViewInstrumentMatch per instrument. If the user's View configuration includes a
-                    # name, this will cause multiple conflicting output streams.
-                    matches.append(
+                if view._match(instrument):
+                    view_instrument_matches.append(
                         _ViewInstrumentMatch(
-                            name=view.name or instrument.name,
-                            resource=self._sdk_config.resource,
-                            instrumentation_info=None,
-                            aggregation=view.aggregation,
+                            name=view._name or instrument.name,
                             unit=instrument.unit,
-                            description=view.description,
+                            description=(
+                                view._description or instrument.description
+                            ),
+                            aggregation=instrument._aggregation,
+                            instrumentation_info=(
+                                instrument.instrumentation_info
+                            ),
+                            resource=self._sdk_config.resource,
+                            attribute_keys=view._attribute_keys,
                         )
                     )
 
             # if no view targeted the instrument, use the default
-            if not matches:
-                # TODO: the logic to select aggregation could be moved
-                if isinstance(instrument, Counter):
-                    agg = SumAggregation(True, AggregationTemporality.DELTA)
-                elif isinstance(instrument, Histogram):
-                    agg = ExplicitBucketHistogramAggregation()
-                else:
-                    agg = LastValueAggregation()
-                matches.append(
+            if not view_instrument_matches:
+                view_instrument_matches.append(
                     _ViewInstrumentMatch(
-                        resource=self._sdk_config.resource,
-                        instrumentation_info=None,
-                        aggregation=agg,
+                        name=instrument.name,
                         unit=instrument.unit,
                         description=instrument.description,
-                        name=instrument.name,
+                        aggregation=instrument._aggregation,
+                        instrumentation_info=instrument.instrumentation_info,
+                        resource=self._sdk_config.resource,
+                        attribute_keys=set()
```

Comment:
```suggestion
                        attribute_keys=None
```

Reply:
I am using an instance of `set` here because the rest of the parameters of `_ViewInstrumentMatch` are mandatory as well.